core/templates/domain/question/QuestionObjectFactory.ts

```diff
@@ -17,149 +17,175 @@
  * question domain objects.
  */
 
-require('domain/state/StateObjectFactory.ts');
-
-angular.module('oppia').factory('QuestionObjectFactory', [
-  'StateObjectFactory', 'DEFAULT_LANGUAGE_CODE', 'INTERACTION_SPECS',
-  function(StateObjectFactory, DEFAULT_LANGUAGE_CODE, INTERACTION_SPECS) {
-    var Question = function(id, stateData, languageCode, version,
-        linkedSkillIds) {
-      this._id = id;
-      this._stateData = stateData;
-      this._languageCode = languageCode;
-      this._version = version;
-      this._linkedSkillIds = linkedSkillIds;
-    };
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+
+import cloneDeep from 'lodash/cloneDeep';
+
+import { InteractionSpecsConstants } from 'pages/interaction-specs.constants';
+import { StateObjectFactory, State } from 'domain/state/StateObjectFactory';
+
+export interface IQuestionBackendDict {
+  id: string;
+  'question_state_data': State;
+  'language_code': string;
```

Comment:
@bansalnitish because linter allows only camel cased props

Reply:
Hmm this seems a bit off, shouldn't we either update the linter to allow this (if this is what its supposed to be), or otherwise update these properties to follow what the linter expects?

Reply:
@vinitamurthi If we consider allowing the linter to use this then we don't need to change the rest of the files unless we would change it to camel-cased props then I would suggest filing a separate issue for this as it would influence other files too I think.

Reply:
Hmm but we already have a pattern of disabling the linter when necessary (for example, in `UpgradedServices`), so let's just do something like:

```typescript
/* eslint-disable camelcase */
question_state_data: State;
/* eslint-enable camelcase */